esp/services/esdl_svc_engine/esdl_store.cpp

```diff
@@ -64,14 +64,14 @@ class CEsdlSDSStore : implements IEsdlStore, public CInterface
         if (!conn)
            throw MakeStringException(-1, "Unable to connect to ESDL Service definition information in dali '%s'", xpath.str());
 
-        conn->close(); //release lock right away
-
         IPropertyTree * deftree = conn->queryRoot();
         if (!deftree)
            throw MakeStringException(-1, "Unable to open ESDL Service definition information in dali '%s'", xpath.str());
 
         //There shouldn't be multiple entries here, but if so, we'll use the first one
         toXML(deftree, esxdl, 0, 0);
+
+        conn->close(); //release lock right away
```

Comment:
Does this gain anything?  The lock will be released as soon as the conn object goes out of scope.

Reply:
No, and dangerous, but I see it's now been removed.